### DISCUSSION TITLE
How to use led that indicates all incoming midi messages?

### DISCUSSION THREAD

**Discussion by kentforth** [tttapa/Control-Surface#432] [General]
I have one led. I want it to react to all incoming midi messages whether it be midi notes from buttons or midi cc from potentiometers. It is like small led on arduino bord indicating all incoming data.
**How can I do that?**
**Is it possible to implement it to library?**

**Comment by tttapa** [comment]
Add a [callback](https://tttapa.github.io/Control-Surface-doc/Doxygen/d1/d5d/MIDI-Input_8ino-example.html) to the MIDI interface you're using, turn the LED on in that callback and record the current time using `millis()`. In the main loop, turn off the LED if x milliseconds have passed since the last MIDI message was received.

**Reply by kentforth** [reply]
I don't know how to add callback, can you give me and example?

**Reply by tttapa** [reply]
See the link in my previous message.

**Reply by kentforth** [reply]
Thank you!
